src/chess/chess_game.rs

```diff
@@ -185,13 +186,25 @@ fn chess_color_to_game_color(c: chess::Color) -> GameColor {
 
 impl PartialEq for ChessPosition {
     fn eq(&self, other: &Self) -> bool {
-        self.board == other.board

```

Comment:
what's the difference?

Reply:
Implementation of PartialEq trait of board is implemented 'defaulty' by the library, meaning the two structs are compared byte by byte. I think this does work correctly, but there are some auxiliary fields which doesn't give any new information, only supplying fast quering answering of some function, and these fields are compared too.
This change may be unnecessary